velox/exec/tests/utils/TpchQueryBuilder.cpp

```diff
@@ -153,6 +155,82 @@ TpchPlan TpchQueryBuilder::getQ1Plan() const {
   return context;
 }
 
+TpchPlan TpchQueryBuilder::getQ4Plan() const {
+  static const std::string kOrders = "orders";
+  static const std::string kLineitem = "lineitem";
+
+  std::vector<std::string> lineitemColumns = {
+      "l_orderkey", "l_commitdate", "l_receiptdate"};
+  std::vector<std::string> ordersColumns = {
+      "o_orderdate", "o_orderkey", "o_orderpriority"};
+
+  auto lineitemSelectedRowType = getRowType(kLineitem, lineitemColumns);
+  const auto& lineitemFileColumns = getFileColumnNames(kLineitem);
+  auto ordersSelectedRowType = getRowType(kOrders, ordersColumns);
+  const auto& ordersFileColumns = getFileColumnNames(kOrders);
+
+  // dateVal :2 = '1993-07-01'
+  const auto orderDate = "o_orderdate";
+  std::string orderDateFilter;
+
+  // DWRF does not support Date type. Use Varchar instead.
+  if (ordersSelectedRowType->findChild(orderDate)->isVarchar()) {
+    orderDateFilter = "o_orderdate between '1993-07-01' and '1993-09-30'";
+  } else {
+    orderDateFilter =
+        "o_orderdate between '1993-07-01'::DATE and '1993-09-30'::DATE";
+  }
+
+  auto planNodeIdGenerator = std::make_shared<PlanNodeIdGenerator>();
+  core::PlanNodeId lineitemPlanNodeId;
+  core::PlanNodeId ordersPlanNodeId;
+
+  auto lineitems = PlanBuilder(planNodeIdGenerator)
+                       .tableScan(
+                           kLineitem,
+                           lineitemSelectedRowType,
+                           lineitemFileColumns,
+                           {},
+                           "l_commitdate < l_receiptdate")
+                       .capturePlanNodeId(lineitemPlanNodeId)
+                       .planNode();
+
+  auto plan =
+      PlanBuilder(planNodeIdGenerator)
+          .localPartition(
+              {},
+              {PlanBuilder(planNodeIdGenerator)
+                   .tableScan(
+                       kOrders,
+                       ordersSelectedRowType,
+                       ordersFileColumns,
+                       {orderDateFilter},
+                       "")
+                   .capturePlanNodeId(ordersPlanNodeId)
+                   .hashJoin(
+                       {"o_orderkey"},
+                       {"l_orderkey"},
+                       lineitems,
+                       "",
+                       {"o_orderpriority"},
+                       core::JoinType::kSemi)
+                   .partialAggregation(
+                       {"o_orderpriority"}, {"count(0) AS partial_order_count"})
+                   .planNode()})
+          .singleAggregation(
+              {"o_orderpriority"}, {"sum(partial_order_count) AS order_count"})
+          .orderBy({"o_orderpriority"}, false)
+          .project({"o_orderpriority", "order_count"})
```

Comment:
This project is redundant and can be removed.

Reply:
Ok sure, thanks